db/migrate/20191128113434_add_hashed_api_key.rb

```diff
@@ -0,0 +1,48 @@
+# frozen_string_literal: true
+class AddHashedApiKey < ActiveRecord::Migration[6.0]
+  def up
+    add_column(:api_keys, :key_hash, :string)
+    add_column(:api_keys, :truncated_key, :string)
+
+    execute(
+      <<~SQL
+        UPDATE api_keys
+        SET truncated_key = LEFT(key, 4)
+      SQL
+    )
+
+    batch_size = 100
+    begin
+      batch = ActiveRecord::Base.connection.select_all <<-SQL
+        SELECT id, key
+        FROM api_keys
+        WHERE key_hash IS NULL
+        LIMIT #{batch_size}
+      SQL
+
+      for row in batch
+        hashed = Digest::SHA256.hexdigest row["key"]
```

Comment:
With mini sql you get row.key, can sha256 be done direct in pg?

Reply:
> can sha256 be done direct in pg

Not without loading the pgcrypto module, which we don't currently use

Reply:
K so we can do a larger batches and we can batch update table, just to minimise db trips during migration, cause this can lock stuff up